### ISSUE TITLE
Create base class to handle copying files for persistence managers

### ISSUE THREAD

**Issue by codymccoy** [HydrologicEngineeringCenter/HEC-FDA#489]
The only persistence managers that remain are the ones that have to do special logic for moving files from the user selected directory into the study directory.
![image](https://user-images.githubusercontent.com/47757848/184363873-7c9dde94-02ad-4b07-86c4-f159bbcb2f0e.png)

They all do similar things: copy single file, copy all files at a certain level, copy whole directory, copy files with certain extensions. The terrains do the copy on a background thread. We might want the hydros to be on a background thread as those can take a while. I think it would be a nice improvement to have most or all of this logic in a "SavesFilesToStudyPersistenceManager" base class.

**Comment by codymccoy** [comment]
for some of these, like the hydros, the copying is being done in the editor vm. I don't like this, that should be the job of the manager.

**Comment by codymccoy** [comment]
This is partially completed. I am not sure there is much benefit in spending more time on this. I moved some of the copying and deleting logic into a static class called StudyFileManager. I created an interface (that doesn't have anything in it) called "IHaveStudyFiles". This allowed me to move the deleting logic into one place:
![image](https://user-images.githubusercontent.com/47757848/186966304-bdf8e462-539e-4f09-b399-1570a69b05e2.png)
